vignettes/in-packages.Rmd

```diff
@@ -0,0 +1,399 @@
+---
+title: "In packages"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Usage and migration}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+## Introduction
+
+This vignette serves two distinct, but related, purposes:
+
+* It documents general best practices for using tidyr in a package, 
+  inspired by [using ggplot2 in packages][ggplot2-packages].
+  
+* It describes migration patterns for the transition from tidyr v0.8.3 to 
+  v1.0.0. This release includes breaking changes to `nest()` and `unnest()`
+  in order to increase consistency within tidyr and with the rest of the
+  tidyverse.
+
+Before we go on, we'll attach the packages we use, expose the version of tidyr, and make a small dataset to use in examples.
+
+```{r setup}
+library(tidyr)
+library(dplyr, warn.conflicts = FALSE)
+library(purrr)
+
+(mini_iris <- iris %>% 
+    as_tibble() %>% 
+    .[c(1, 2, 51, 52, 101, 102), ])
+```
+
+## Using tidyr in packages
+
+Compared to dplyr and ggplot2, most tidyr functions have __select semantics__, not the more common mutate semantics. This means that you typically provide tidyr functions with an expression that works with column names (like `starts_with("x")` or `a:c`) not with column values (like `y = x * 2`). As a consequence, it's worthwhile to clarify the best patterns for calling tidyr functions inside another package.
```

Comment:
> not the more common mutate semantics

Mutate semantics are also about vectorisation, and sequential updating. Should this be "action" semantics instead?